languageserver/integration/integration.go

```diff
@@ -111,14 +279,68 @@ func (i *FlowIntegration) initialize(initializationOptions any) error {
 		return nil
 	}
 
-	// Load the config state if provided
+	// Load the config state if provided. If it fails, don't fail initialization;
+	// still set the init-config override so the server can run and the user can fix the file.
 	configPath = cleanWindowsPath(configPath)
 	err := i.state.Load(configPath)
 	if err != nil {
-		return err
+		// Seed cfgManager with the init-config path so project scoping still works
+		if i.cfgManager != nil {
+			i.cfgManager.enableFlowClient = i.enableFlowClient
+			i.cfgManager.numberOfAccounts = 0
+			i.cfgManager.SetInitConfigPath(configPath)
+			// Record the failure in cfgManager by attempting to resolve state there as well
+			_, _ = i.cfgManager.ResolveStateForProject(configPath)
+		}
+		// Try to parse enough of flow.json to detect invalid contract paths and surface to the user via ShowMessage
+		if i.client != nil {
+			// best-effort: read file and try to detect bad paths
+			if data, readErr := i.loader.ReadFile(configPath); readErr == nil {
+				var parsed struct {
+					Contracts map[string]string `json:"contracts"`
+				}
+				if jsonErr := json.Unmarshal(data, &parsed); jsonErr == nil && len(parsed.Contracts) > 0 {
+					for _, rel := range parsed.Contracts {
+						candidate := filepath.Join(filepath.Dir(configPath), rel)
+						if _, statErr := i.loader.Stat(candidate); statErr != nil {
+							// Inform user once per root
+							root := filepath.Dir(configPath)
+							i.invalidWarnedMu.Lock()
+							if _, seen := i.invalidWarnedRoots[root]; !seen {
+								i.invalidWarnedRoots[root] = struct{}{}
+								i.invalidWarnedMu.Unlock()
+								// send message via client logger if available
+								// Note: we don't have conn here; rely on code lenses/diagnostics path to surface errors on open
+								// Alternatively, no-op. Keeping mark prevents repeated errors.
+							} else {
+								i.invalidWarnedMu.Unlock()
+							}
```

Comment:
Same as above: Instead of carefully having to ensure the lock is unlocked in all execution paths, use a `defer` to have the unlock done automatically. Depending on how long the lock should hold / what the critical section is, wrap in an immediately invoked function. Also consider refactoring this into a separate function which is easier to reason about (this function is already massive)